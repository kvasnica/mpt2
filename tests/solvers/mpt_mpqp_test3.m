function mpt_mpqp_test3

% tests that we properly handle cases where the [G W] matrix contains multiple
% identical rows (if so, we have a problem when doing matrix inverse). fixed by:
% http://control.ee.ethz.ch/~mpt/hg/mpt?cs=47dc10d52961

load mpqp_redundant_prob
% compute feasible set by projection
Pfeas = polytope([-Matrices.E Matrices.G], Matrices.W);
Pproj = projection(Pfeas, 1:size(Matrices.E, 2));

fprintf('Degeneracy should be indicated:\n\n');
Pn=mpt_mpqp(Matrices);
mbg_assertequal(length(Pn), 1);

mbg_asserttrue(Pn==Pproj); % test that mpqp returned correct solution
