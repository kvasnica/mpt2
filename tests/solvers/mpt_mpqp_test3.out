Degeneracy should be indicated:

mpt_mpqp: Degeneracy
mpt_mpqp: 1 regions